packages/eslint-config/index.js

```diff
@@ -30,6 +30,7 @@ module.exports = {
     parser: "@typescript-eslint/parser",
     parserOptions: {
         sourceType: "module",
+        createDefaultProgram: true,
```

Comment:
this ended up causing a huge performance hit (see https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/parser#parseroptionscreatedefaultprogram), so I am reverting this change in https://github.com/palantir/blueprint/pull/4078